### DISCUSSION TITLE
Update the entier user with the user Auth API ?

### DISCUSSION THREAD

**Discussion by Yberion** [zitadel/zitadel#7786] [Q&A] [answered]
Hello,

Is there a way to update the entire user?

My point would be to update everything at once, all info from this page:

![image](https://github.com/zitadel/zitadel/assets/4186385/0db24123-2833-4932-868a-2451ce1c42a1)

Something similar to https://zitadel.com/docs/apis/resources/auth/auth-service-get-my-user but with a PUT, this makes little sens to me to make 4 requests or more to update the entire profile.

This could be something similar to this endpoint https://zitadel.com/docs/apis/resources/user_service/user-service-update-human-user, maybe we can use the user token on this endpoint for his own userId?

**Comment by hifabienne** [comment, marked as answer]
hei @Yberion 
The second endpoint you have sent is exactly for that.
Our goal with our new API is to have one endpoint where each user can do the call. Based on your permissions you are allowed or not.
This is the different to the APIs before where we had an API for the managers, and an API for the authenticated user itself.

**Reply by Yberion** [reply]
Oh, that's awesome! I though those endpoints were for manager only 😄 

I'm a bit confused by this endpoint https://zitadel.com/docs/apis/resources/user_service/user-service-update-human-user

It asks for a lot of required fields, but after testing, it seems that only `profile.familyName` and `profile.givenName` are required in this endpoint.

![image](https://github.com/zitadel/zitadel/assets/4186385/d9932746-8633-458e-8d6f-aa8b9524b0c3)

Should I create a bug issue?

**Reply by hifabienne** [reply]
I am not sure, but I think its a docs problem.
The email, password etc are submessages.
If you send them the fields are required, but if you do not send it is not requried.
@eliobischof @stebenz am I right? or do I miss understand it?

**Reply by Yberion** [reply]
To followup, indeed, if I set the `email.email` I don't need to set `profile.familyName` and `profile.givenName`, this seems unclear in the doc 😸

**Reply by Yberion** [reply]
Another thing on why I said `I though those endpoints were for manager only`, it is because of some endpoints, such as https://zitadel.com/docs/apis/resources/user_service/user-service-list-users, this endpoint seems to be for managers, I don't get why a regular user could search for other users, this would be problematic in term of GDPR.

On the V1 doc, this felt more clear as it was a bit more separated.

But that's another topic and it's probably because of my lack of knowledge ^^

**Reply by stebenz** [reply]
Yes, the docs seem a little confusing, for example `profile` is optional but if you set `profile` there has to be a `profile.givenName`.

The user list endpoint is permission checked that you only get returned the users you have permission to see, which can also be limited to only your user.
